app/src/main/java/com/github/studydistractor/sdp/ui/CreateEventScreen.kt

```diff
@@ -0,0 +1,287 @@
+                    modifier = Modifier
+                        .padding(vertical = 32.dp)
+                        .testTag("create-event-screen__title")
+                )
+            }
+
+            CreateActivityNameAndDescriptionFields(
+                createActivityViewModel = createEventViewModel
+            )
+
+            CreateDateTimeField(
+                valueText = uiState.startDateTime,
+                onValueChange = { createEventViewModel.updateStartDateTimeText(it) },
+                labelText = "Start date and time (dd-mm-yyyy hh:mm)",
+                onClickButton = { startDateTimePicker.show() }
+            )
+
+            CreateDateTimeField(
+                valueText = uiState.endDateTime,
+                onValueChange = { createEventViewModel.updateEndDateTimeText(it) },
+                labelText = "End date and time (dd-mm-yyyy hh:mm)",
+                onClickButton = { endDateTimePicker.show() }
+            )
+
+            CreateLatitudeAndLongitudeFields(
+                createActivityViewModel = createEventViewModel,
+                uiState = uiState
+            )
+
+            PointsAwardedField(createActivityViewModel = createEventViewModel, uiState = uiState)
+            LateParticipationCheckbox(
+                createEventViewModel = createEventViewModel,
+                uiState = uiState
+            )
+
+            CreateActivityButton(
+                createActivityViewModel = createEventViewModel,
+                onActivityCreated = onEventCreated,
+                buttonText = "Create event"
+            )
+        }
+    }
+}
+
+/**
+ * Creates a date picker dialog
+ * @param context Context
+ * @param setDateTime Callback to be called when date and time is set
+ * @param year Year
+ * @param month Month
+ * @param dayOfMonth Day of month
+ * @param hour Hour
+ * @param minute Minute
+ * @return DatePickerDialog
+ */
+private fun dateTimePickerDialog(
+    context: Context,
+    setDateTime: (year: Int, month: Int, day: Int, h: Int, m: Int) -> Unit,
+    year: Int,
+    month: Int,
+    dayOfMonth: Int,
+    hour: Int,
+    minute: Int
+): DatePickerDialog {
+    return DatePickerDialog(
+        context,
+        { _, y, m, d ->
+            val timePicker = TimePickerDialog(
+                context,
+                { _, h, min -> setDateTime(y, m, d, h, min) },
+                hour,
+                minute,
+                true
+            )
+            timePicker.show()
+        },
+        year,
+        month,
+        dayOfMonth
+    )
+}
+
+/**
+ * Creates a text field to allow user to enter date and time
+ * @param valueText Text to be displayed in the text field
+ * @param onValueChange Callback to be called when text is changed
+ * @param labelText Label text
+ * @param onClickButton Callback to be called when button is clicked
+ * @return TextField
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateDateTimeField(
```

Comment:
Should be a component?

Reply:
There's the risk of duplicating it without noticing: we can't expect people to check all the other screens for something, but we can expect them to take a look into our components package. 

As this is not important imo, feel free to resolve this comment without implementing.